In [1]:
import pandas as pd

# Зчитуємо CSV файл
df = pd.read_csv('bestsellers with categories.csv')


In [3]:
#Виведіть перші 10 рядків (метод head):
print(df.head(10))


                                                Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   
5      A Dance with Dragons (A Song of Ice and Fire)   
6  A Game of Thrones / A Clash of Kings / A Storm...   
7                     A Gentleman in Moscow: A Novel   
8      A Higher Loyalty: Truth, Lies, and Leadership   
9                          A Man Called Ove: A Novel   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fic

In [4]:
#Виведіть розміри датасету (метод shape):
print("Розмір датасету:", df.shape)


Розмір датасету: (550, 7)


# Питання: Про скільки книг зберігає дані датасет?
# Відповідь: Датасет містить інформацію про 550 книг.

In [5]:
#Змініть назви всіх колонок: назва з маленької букви, без пробілів через підкреслення. Вивести кількість пропусків у кожному стовпчику.
# Змінюємо назви стовпців та виводимо кількість пропусків
df.columns = ['name', 'author', 'user_rating', 'reviews', 'price', 'year', 'genre']
print(df.isna().sum())


name           0
author         0
user_rating    0
reviews        0
price          0
year           0
genre          0
dtype: int64


# Питання: Чи є в якихось змінних пропуски? (Так / ні)
# Відповідь: Ні, в жодному зі стовпців немає пропусків.

In [6]:
#За допомогою методу unique перевірте, які є унікальні жанри в стовпці "genre".

# Виводимо унікальні жанри
unique_genres = df['genre'].unique()
print("Унікальні жанри:", unique_genres)


Унікальні жанри: ['Non Fiction' 'Fiction']


In [7]:
# Визначте максимальну, мінімальну, медіанну та середню ціну за допомогою методів max, min, mean, median.

# Визначаємо максимальну, мінімальну, медіанну та середню ціну
max_price = df['price'].max()
min_price = df['price'].min()
mean_price = df['price'].mean()
median_price = df['price'].median()

print("Максимальна ціна:", max_price)
print("Мінімальна ціна:", min_price)
print("Середня ціна:", mean_price)
print("Медіанна ціна:", median_price)


Максимальна ціна: 105
Мінімальна ціна: 0
Середня ціна: 13.1
Медіанна ціна: 11.0


In [8]:
#Виконайте наступні завдання з пошуку і сортування даних:

# Питання: Який рейтинг у датасеті найвищий?
highest_rating = df['user_rating'].max()
print("Найвищий рейтинг:", highest_rating)

# Питання: Скільки книг мають такий рейтинг?
count_highest_rating = df[df['user_rating'] == highest_rating]['user_rating'].count()
print("Кількість книг з найвищим рейтингом:", count_highest_rating)

# Питання: Яка книга має найбільше відгуків?
book_with_most_reviews = df[df['reviews'] == df['reviews'].max()]['name'].values[0]
print("Книга з найбільшою кількістю відгуків:", book_with_most_reviews)

# Питання: З тих книг, що потрапили до Топ-50 у 2010 році, яка книга найдорожча?
most_expensive_2010 = df[(df['year'] == 2010)]['price'].max()
print("Найдорожча книга у 2010 році:", most_expensive_2010)

# Питання: Скільки книг жанру Fiction потрапили до Топ-50 у 2012 році
count_fiction_2012 = df[(df['year'] == 2012) & (df['genre'] == 'Fiction')]['name'].count()
print("Кількість книг жанру Fiction у 2012 році:", count_fiction_2012)

# Питання: Скільки книг з рейтингом 4.9 потрапило до рейтингу у 2010 та 2011 роках
count_49_rating_2010_2011 = df[(df['user_rating'] == 4.9) & ((df['year'] == 2010) | (df['year'] == 2011))]['name'].count()
print("Кількість книг з рейтингом 4.9 у 2010 та 2011 роках:", count_49_rating_2010_2011)

# Вивести максимальну і мінімальну ціну для жанру Fiction і NonFiction
max_min_prices = df.groupby('genre')['price'].agg(['max', 'min'])
print("Максимальні та мінімальні ціни для жанрів Fiction і NonFiction:")
print(max_min_prices)


Найвищий рейтинг: 4.9
Кількість книг з найвищим рейтингом: 52
Книга з найбільшою кількістю відгуків: Where the Crawdads Sing
Найдорожча книга у 2010 році: 46
Кількість книг жанру Fiction у 2012 році: 21
Кількість книг з рейтингом 4.9 у 2010 та 2011 роках: 1
Максимальні та мінімальні ціни для жанрів Fiction і NonFiction:
             max  min
genre                
Fiction       82    0
Non Fiction  105    0
